core/src/main/java/com/linecorp/armeria/client/Http2ResponseDecoder.java

```diff
@@ -257,6 +258,12 @@ public int onDataRead(
 
         if (endOfStream) {
             res.close();
+
+            if (needsToDisconnectNow() && !goAwayHandler.sentGoAway() && !goAwayHandler.receivedGoAway()) {
```

Comment:
How about adding the same logic in `onHeadersRead`?

Reply:
I missed it. I should've added it to them.